eng/packages/http-client-csharp/generator/Azure.Generator/src/Primitives/KnownAzureTypes.cs

```diff
@@ -27,5 +51,29 @@ internal static class KnownAzureTypes
             [AzureLocationId] = typeof(AzureLocation),
             [ArmIdId] = typeof(ResourceIdentifier),
         };
+
+        private static readonly IReadOnlyDictionary<Type, SerializationExpression> _typeToSerializationExpression = new Dictionary<Type, SerializationExpression>
+        {
+            [typeof(Guid)] = SerializeTypeWithImplicitOperatorToString,
+            [typeof(IPAddress)] = SerializeTypeWithToString,
+            [typeof(ETag)] = SerializeTypeWithImplicitOperatorToString,
+            [typeof(AzureLocation)] = SerializeTypeWithImplicitOperatorToString,
+            [typeof(ResourceIdentifier)] = SerializeTypeWithImplicitOperatorToString,
+        };
+
+        private static readonly IReadOnlyDictionary<Type, DeserializationExpression> _typeToDeserializationExpression = new Dictionary<Type, DeserializationExpression>
+        {
+            [typeof(Guid)] = DeserializeNewInstanceStringLikeType,
+            [typeof(IPAddress)] = DeserializeParsableStringLikeType,
+            [typeof(ETag)] = DeserializeParsableStringLikeType,
+            [typeof(AzureLocation)] = DeserializeNewInstanceStringLikeType,
+            [typeof(ResourceIdentifier)] = DeserializeNewInstanceStringLikeType,
+        };
+
+        internal static bool TryGetPrimitiveType(string id, [MaybeNullWhen(false)] out CSharpType type) => _idToTypes.TryGetValue(id, out type);
```

Comment:
Since this class is internal already, the members can just be public like the delegate.

Reply:
Fixed